test/functional/search_controller_test.rb

```diff
@@ -317,6 +319,18 @@ def no_results
     assert_equal "1",       @response.headers["X-Slimmer-Result-Count"]
   end
 
+  test "should pass the search variant through when one is present" do
+    with_variant SearchMatchLength: "B" do
+      get :index, q: "Variant B"
+    end
```

Comment:
`with_variant` checks whether the test name and the variant are in the response header. Should we assert something additional to that?